packages/devtools_app/lib/src/inspector/flutter/inspector_data_models.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import '../enum_deserializer.dart';
+
+const Type boxConstraintsType = BoxConstraints;
+
+Constraints deserializeConstraints(Map<String, Object> json) {
+  // TODO(albertusangga): Support SliverConstraint
+  if (json == null || json['type'] != boxConstraintsType.toString())
+    return null;
+  return BoxConstraints(
+    minWidth: json['minWidth'],
+    maxWidth: json['hasBoundedWidth'] ? json['maxWidth'] : double.infinity,
+    minHeight: json['minHeight'],
+    maxHeight: json['hasBoundedHeight'] ? json['maxHeight'] : double.infinity,
+  );
+}
+
+/// TODO(albertusangga): Move this to [RemoteDiagnosticsNode] once dart:html app is removed
+@immutable
+class RenderFlexProperties {
+  const RenderFlexProperties({
+    this.direction,
+    this.mainAxisAlignment,
+    this.mainAxisSize,
+    this.crossAxisAlignment,
+    this.textDirection,
+    this.verticalDirection,
+    this.textBaseline,
+  });
+
+  final Axis direction;
+  final MainAxisAlignment mainAxisAlignment;
+  final MainAxisSize mainAxisSize;
+  final CrossAxisAlignment crossAxisAlignment;
+  final TextDirection textDirection;
+  final VerticalDirection verticalDirection;
+  final TextBaseline textBaseline;
+  // TODO(albertusangga) Add size & constraints in this class
+
+  static final directionDeserializer = EnumDeserializer<Axis>(Axis.values);
```

Comment:
make these private top level members instead of statics.

Reply:
I think this is better left to be static since technically we can reuse single deserializer object for deserializing multiple `FlexProperties`, or memory is not a concern here?
Just curious